cadence/server/api_actions.go

```diff
@@ -121,85 +130,136 @@ func liquidsoapRequest(path string) (message string, err error) {
 	return message, nil
 }
 
-var preTitle, preArtist, nowTitle, nowArtist string = "-", "-", "-", "-"
-var preHost, preMountpoint, nowHost, nowMountpoint string = "-", "-", "-", "-"
-var preListeners, nowListeners float64 = -1, -1
+func liquidsoapSkip() (message string, err error) {
+	clog.Debug("liquidsoapRequest", "Connecting to liquidsoap service...")
+	conn, err := net.Dial("tcp", c.SourceAddress+c.SourcePort)
+	defer conn.Close()
+	if err != nil {
+		clog.Error("liquidsoapRequest", "Failed to connect to audio source server.", err)
+		return "", err
+	}
+	fmt.Fprintf(conn, "cadence1.skip\n")
+	// Listen for response
+	message, _ = bufio.NewReader(conn).ReadString('\n')
+	fmt.Fprintf(conn, "quit"+"\n")
+	return message, nil
+}
 
-// Takes no arguments.
-// Returns nothing, but sends updated stream info to SSE and sets the global variables.
-// It is launched as a goroutine by init.
+// Watches the music directory (CSERVER_MUSICDIR) for any changes, and reconfigures the database.
+func filesystemMonitor() {
+	watcher, err := fsnotify.NewWatcher()
```

Comment:
Are we considering updating to a recursive method (rfsnotify/roll our own/etc), or are we going to constrain that if you want updates to appear you need to make sure it touches the mtime on the MusicDir?

Reply:
Can you give more detail on the latter?

Reply:
Well, it's my understanding that the `fsnotify` watcher only fires off if the directory we're watching gets modified.

That's fine if the filesystem propagates changes (writing a file into a subdirectory fires a modification event on the parent), or if the administrator knows this is a thing and puts files flat in the music dir, but if an enterprising filesystem decides it doesn't need to be nice and make sure the parent directory gets a "change" when a subdirectory gets a new file, then that would mean using it for the music dir would leave gaps in the populator's awareness of change.

Basically - Should we/might we/is it worth switching to something like `rfsnotify` in the future which handles subdirectories explicitly, and/or is my understanding incorrect?

Reply:
I think I understand now. The target directory is not presently recursively searched during population. Only the root of the target of searched. 

Though, of course, I'm now imagining a user who wants to simply hook up their Cadence installation to their `~/Music/` folder, which is populated with hundreds of subfolders of albums. 

Yeah. We should plan to revisit this eventually and implement both recursive population and monitoring.